### PR TITLE
ddl: fix drop index failed when index contain auto_increment column and the auto_increment column is primary key

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Before this PR:

```sql
mysql>create table t3 (a int auto_increment key, b int, index idx(b,a))
Query OK, 0 rows affected
Time: 0.074s
mysql>alter table t3 drop index idx;   -- should sucessful.
(1075, u'Incorrect table definition; there can be only one auto column and it must be defined as a key')
```

### What is changed and how it works?

Fix DDL drop index with `auto_increament` check function bug.

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects

### Release note 

Fix issue of drop index failed when index contain auto_increment column and the auto_increment column is the primary key.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: ddl/serial_test.go [hunk 1]
```diff
@@ -152,6 +152,24 @@ func (s *testSerialSuite) TestPrimaryKey(c *C) {
...
 }
 
+func (s *testSerialSuite) TestDropAutoIncrementIndex(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	cfg := config.GetGlobalConfig()
+	newCfg := *cfg
+	orignalAlterPrimaryKey := newCfg.AlterPrimaryKey
+	// Only when disable the AlterPrimaryKey config, this test will make sense.
+	newCfg.AlterPrimaryKey = false
```
Comment (zimulala) [inline ddl/serial_test.go]: It seems that the default value is false, could we remove it?

Comment (crazycs520) [reply ddl/serial_test.go]: reasonable.

**Pull request merged**